azavea/echo-locator
# Feature/kak/profile page#22

## Overview

Add page with a form to enter or edit user profile. Modify search to only use voucher number and match exactly. (See discussion on #55).

Also adds header links to the search and profile pages, and clears profile on logout.


### Demo

New profile page:
![image](https://user-images.githubusercontent.com/960264/53996431-5436d680-4106-11e9-8e72-3618d275471c.png)

Search page:
![image](https://user-images.githubusercontent.com/960264/53996473-80eaee00-4106-11e9-9553-fde8a09ee747.png)

On voucher not found:
![image](https://user-images.githubusercontent.com/960264/53996489-96f8ae80-4106-11e9-988f-37147f7951e8.png)


### Notes

The voucher holder name is now an editable field on the profile page, as it is no longer a search/create key.


## Testing Instructions

 * Should be able to edit profile at `/profile`
 * Should be able to add up to three addresses
 * Address geocoder should work as expected
 * Changes should be saved to `localStorage` and to S3 on clicking 'go'
 * Navigating back to profile page after editing should reflect saved changes
 * Clicking 'cancel' should discard changes and navigate to the previous page (search or map)
 * 'Edit' header button should redirect to profile page
 * 'New' header button should redirect to search page
 * Profile should be unset on logout
 * Search should go to profile page on exact voucher number match
 * Search should prompt to create account if voucher number not found


Closes #22 
Closes #31 
Closes #59
